app/oxalis/thirdparty/BrainTracing.scala

```diff
@@ -34,41 +34,42 @@ object BrainTracing extends LazyLogging with FoxImplicits {
   lazy val Mailer =
     Akka.system(play.api.Play.current).actorSelection("/user/mailActor")
 
-  def register(user: User): Future[String] = {
-    // TODO: fix, make team dynamic
-    if (isActive && user.organization == "Connectomics department") {
-      val result = Promise[String]()
-      val brainTracingRequest = WS
-        .url(CREATE_URL)
-        .withAuth(USER, PW, WSAuthScheme.BASIC)
-        .withQueryString(
-          "license" -> LICENSE,
-          "firstname" -> user.firstName,
-          "lastname" -> user.lastName,
-          "email" -> user.email,
-          "pword" -> user.md5hash)
-        .get()
-        .map { response =>
-          result complete (response.status match {
-            case 200 if isSilentFailure(response.body) =>
-              Success("braintracing.error")
-            case 200 =>
-              Success("braintracing.new")
-            case 304 =>
-              Success("braintracing.exists")
-            case _ =>
-              Success("braintracing.error")
-          })
-          logger.trace(s"Creation of account ${user.email} returned Status: ${response.status} Body: ${response.body}")
-        }
-      brainTracingRequest.onFailure{
-        case e: Exception =>
-          logger.error(s"Failed to register user '${user.email}' in brain tracing db. Exception: ${e.getMessage}")
+  def registerIfNeeded(user: UserSQL): Fox[String] =
```

Comment:
@fm3 Did you double check to change all `register` calls to `registerIfNeeded`? Also here, maybe just use an inner function, if the new `register` should actually not be called.

Reply:
I made register private, forbidding external calls to it

Reply:
True, didn't notice this!